lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -440,6 +440,13 @@ resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE, DeclContext *DC) {
   DeclName Name = UDRE->getName();
   SourceLoc Loc = UDRE->getLoc();
 
+  // 'Any' is the empty protocol composition; resolve it before doing an
+  // unqualified lookup
+  if (Name.getBaseName() == Context.Id_Any) {
```

Comment:
We do lose people who wrote `Swift.Any`, but we could get that back with a top-level typealias declared in a similar way.